src/main/java/core/basesyntax/strategy/ReturnHandler.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.FruitTransaction;
+import java.util.List;
+
+public class ReturnHandler implements OperationHandler {
+    private static final String RETURN = "r";
+
+    @Override
+    public List<FruitTransaction> getFruitFinalQuantity(List<FruitTransaction> listOfAll,
+                                                        List<FruitTransaction> listOfFruits) {
+        for (FruitTransaction listOfFruit : listOfFruits) {
+            int oldQuantity = 0;
+            for (FruitTransaction fruitTransaction : listOfAll) {
+                if (listOfFruit.getFruit().equals(fruitTransaction.getFruit())
+                        && fruitTransaction.getType().equals(RETURN)) {
+                    int quantity = fruitTransaction.getQuantity();
+                    listOfFruit.setQuantity(quantity + oldQuantity);
+                    oldQuantity = quantity;
+                }
+            }
+        }
+        return listOfFruits;
+    }
+}
```

Comment:
The `ReturnHandler` class should not be responsible for both updating and retrieving the final quantity of fruits. This violates the Single Responsibility Principle (one of the SOLID principles). Consider splitting the responsibilities into separate classes or methods.